test/hotspot/jtreg/compiler/loopopts/superword/ReductionPerf.java

```diff
@@ -21,242 +21,568 @@
  * questions.
  */
 
-/**
+/*
  * @test
- * @bug 8074981
- * @summary Add C2 x86 Superword support for scalar product reduction optimizations : int test
- * @requires os.arch=="x86" | os.arch=="i386" | os.arch=="amd64" | os.arch=="x86_64" | os.arch=="aarch64" | os.arch=="riscv64"
- *
- * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions
- *      -XX:LoopUnrollLimit=250 -XX:CompileThresholdScaling=0.1
- *      -XX:CompileCommand=exclude,compiler.loopopts.superword.ReductionPerf::main
- *      -XX:+SuperWordReductions
- *      compiler.loopopts.superword.ReductionPerf
- * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions
- *      -XX:LoopUnrollLimit=250 -XX:CompileThresholdScaling=0.1
- *      -XX:CompileCommand=exclude,compiler.loopopts.superword.ReductionPerf::main
- *      -XX:-SuperWordReductions
- *      compiler.loopopts.superword.ReductionPerf
+ * @bug 8074981 8302652
+ * @summary Test SuperWord Reduction Perf.
+ * @requires vm.compiler2.enabled
```

Comment:
This is not enough. Yes, we need to check for C2 presence. But you need skip `arm`, `ppc` and `s390` which have C2. You need second `@requires` as original but you can reduce checks for x86:
```
 * @requires vm.simpleArch == "x86" | vm.simpleArch == "x64" | vm.simpleArch == "aarch64" | vm.simpleArch == "riscv64""
```